File: app/models/weather.rb
class Weather < ApplicationRecord
  def get_weather(lat, lon)
    response = HTTParty.get("https://api.openweathermap.org/data/2.5/weather?lat=#{lat}&lon=#{lon}&appid=#{Rails.application.credentials.weather_api_key}")
  end
end
# class Weather
#   def initialize(lat, lon)
#     @lat = lat
#     @lon = lon
#   end
#
#   def get_weather
#     response = HTTParty.get('api.openweathermap.org/data/2.5/weather?lat={lat}&lon={lon}&appid=weather_api_key')
#     response["description"]["temperature"]
#   end
# end
#

File: app/assets/javascripts/weather.js
class Weather {
  constructor(lat, lon) {
    this.lat = lat;
    this.lon = lon;
  }
}

async getWeatherByCity(lat, lon) {
    try {
      let response = await fetch(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&appid=${Rails.application.credentials.weather_api_key}`);
      let jsonifiedResponse = await response.json();
      return jsonifiedResponse;
    } catch(error) {
      console.error("There was an error handling your request: " + error.message);
    }
  }
